executable/microsoft_pe.ksy

```diff
@@ -408,3 +413,134 @@ types:
         size: length - 8
         doc: Contains a certificate, such as an Authenticode signature.
     doc-ref: 'https://docs.microsoft.com/en-us/windows/desktop/debug/pe-format#the-attribute-certificate-table-image-only'
+
+  resource_directory_table:
+    params:
+      - id: depth
+        type: u4
+      - id: section_file_offset
+        type: u4
+      - id: section_virtual_address
+        type: u4
+    seq:
+      # This field is reserved for future use. It is currently set to zero.
+      - id: characteristics
+        type: u4
+      - id: time_date_stamp
+        type: u4
+        -orig-id: 'TimeDateStamp'
+      - id: version
+        type: version_u2
+      - id: num_named_entries
+        type: u2
+        -orig-id: NumberOfNamedEntries
+      - id: num_id_entries
+        type: u2
+        -orig-id: NumberOfIdEntries
+      - id: named_entries
+        type: resource_directory_entry(depth)
+        repeat: expr
+        repeat-expr: num_named_entries
+      - id: id_entries
+        type: resource_directory_entry(depth)
+        repeat: expr
+        repeat-expr: num_id_entries
+    doc-ref: 'https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#resource-directory-table'
```

Comment:
Please move this `doc-ref` key on the top of the type definition as per the [style guide](https://doc.kaitai.io/ksy_style_guide.html#type).

Putting the `doc` and `doc-ref` below `seq`  or `instances` makes the KSY confused most of the time, see https://github.com/kaitai-io/kaitai_struct_formats/pull/303#discussion_r454140325 for a more extreme example.

Reply:
Done, i've also changed a couple of other places in this PR